packages/core/database/src/dialects/mysql/index.ts

```diff
@@ -58,21 +58,29 @@ export default class MysqlDialect extends Dialect {
         .raw(`set session sql_require_primary_key = 0;`)
         .connection(nativeConnection);
     } catch (err) {
-      // Ignore error due to lack of session permissions
+      if ((err as any)?.code === 'ER_SPECIFIC_ACCESS_DENIED_ERROR') {
+        console.warn('Failed to set session sql_require_primary_key = 0');
+      } else {
+        throw err;
+      }
     }
 
     // We only need to get info on the first connection in the pool
     if (!this.info) {
-      this.info = await this.databaseInspector.getInformation();
+      this.info = await this.databaseInspector.getInformation(nativeConnection);
     }
   }
 
   async startSchemaUpdate() {
+    await this.db.connection.raw(`set foreign_key_checks = 0;`);
     try {
-      await this.db.connection.raw(`set foreign_key_checks = 0;`);
       await this.db.connection.raw(`set session sql_require_primary_key = 0;`);
     } catch (err) {
-      // Ignore error due to lack of session permissions
+      if ((err as any)?.code === 'ER_SPECIFIC_ACCESS_DENIED_ERROR') {
```

Comment:
if there is an error my app was just waiting until knex timeout and then failed to start so why not throw early ?

Reply:
I think this error was a red herring. What was happening was that this error would get thrown, then it would proceed to getInformation, which called .raw, which opened a new thread on the pool, which needed to initialize itself, which then called getInformation again....

Reply:
and I don't think we need this query that is throwing to begin with. I have no idea why it was added, but as far as I can tell everything in strapi seems to work with mysql without it working, that's why we were suppressing it. But why have it if it's not needed?

Reply:
I'm pretty sure it's to support a edge case like a mysql hosted DB somewhere. Would need to find when it was 1st introduced

Reply:
That's why my inclination was to leave it alone, better than breaking something. Do you think I should remove the .warn so it doesn't start showing up for everyone who runs without SUPER privs?